worktree.go

```diff
@@ -469,8 +469,11 @@ func (w *Worktree) resetWorktree(t *object.Tree, files []string) error {
 			}
 		}
 
-		if err := w.checkoutChange(ch, t, b); err != nil {
-			return err
+		status, err := w.Status()
```

Comment:
Calling status can be an [expensive](#181) operation, so we should avoid calling it in a loop. Please move `L472` above the `for` statement.

The error handling would have to be updated accordingly.

Reply:
Done! And it is noticeably faster :)